src/Util/Common.hs

```diff
@@ -58,4 +60,54 @@ gameObjects game = map playerObject (gamePlayers game)
 
 
 updateObjects :: Game -> Game
-updateObjects game = foldr (\obj -> objectOnUpdate obj $ obj) game (gameObjects game)
\ No newline at end of file
+updateObjects game = foldr (\obj -> stringToUpdateFunction (objectOnUpdateName obj) $ obj) game (gameObjects game)
+
+activatePlayer :: String -> Object -> Game -> Game
+activatePlayer name object game =
+  foldr (\player -> (stringToActivateFunction (objectOnActivateName . playerObject $ player))
+                    (objectsCollide object (playerObject player))
+                    (playerObject player)
+        )
+        game
+        (filter isTarget (gamePlayers game))
+  where isTarget player = (objectName . playerObject $ player) == name
+
+namedUpdateFunctions :: [NamedUpdateFunction]
+namedUpdateFunctions = [("onUpdateMario", activatePlayer "luigi")]
+
+namedActivateFunctions :: [NamedActivateFunction]
+namedActivateFunctions = [("resizeSelf", resizeSelf)]
+
+stringToUpdateFunction :: String -> Object -> Game -> Game
+stringToUpdateFunction name = let namedFunction = find (\(x,_) -> x == name) namedUpdateFunctions
+                              in if isJust namedFunction
+                                 then snd (fromJust namedFunction) 
+                                 else (\_ -> id) 
+
+stringToActivateFunction :: String -> Bool -> Object -> Game -> Game
+stringToActivateFunction name = let namedFunction = find (\(x,_) -> x == name) namedActivateFunctions
+                                in if isJust namedFunction
+                                   then snd (fromJust namedFunction) 
+                                   else (\_ _ -> id)  
```

Comment:
Generalize plz
```haskell
findByName :: String -> [(String, a)] -> a
```

Reply:
@dm-gh a function with this signature cannot exist in Haskell (it'll be a partial function, undefined for empty list). It should be instead something like:

```haskell
-- | Lookup a value by key in an associative list.
-- Return default value when the key is not in the list.
-- Return the first value if list has duplicate keys.
lookupWithDefault
  :: v        -- ^ Default value (for when key is not found).
  -> k        -- ^ Key to lookup.
  -> [(k, v)] -- ^ A list of key-value pairs.
  -> v
```